onnx/backend/test/cpp/driver/test_driver.cc

```diff
@@ -199,25 +198,62 @@ ResolvedTestCase LoadSingleTestCase(const UnsolvedTestCase& t) {
   st.test_case_name_ = t.test_case_name_;
   ONNX_NAMESPACE::ParseProtoFromBytes(
       &st.model_, raw_model.c_str(), raw_model.size());
-
+  int test_data_counter = 0;
+  std::string tensor_str = "Tensor";
+  std::string sequence_str = "Sequence";
+  std::string map_str = "Map";
   for (auto& test_data : t.test_data_) {
     ResolvedTestData proto_test_data;
-
     for (auto& input_file : test_data.input_filenames_) {
       std::string input_data;
+      ONNX_NAMESPACE::ValueInfoProto input_info;
       LoadSingleFile(input_file, input_data);
-      ONNX_NAMESPACE::TensorProto input_proto;
-      ONNX_NAMESPACE::ParseProtoFromBytes(
-          &input_proto, input_data.c_str(), input_data.size());
-      proto_test_data.inputs_.emplace_back(std::move(input_proto));
+      input_info = st.model_.graph().input(test_data_counter);
+      if(input_info.doc_string().find(tensor_str) != std::string::npos) {
```

Comment:
curious why are we able to determine input type through `doc_string`? Is there some part in code that we are setting 'Sequence/Tensor/Map' in `doc_string`?

Reply:
It prints out with the type in it, so I was using that as a metric, but I agree -- it's an imperfect solution since it's not actually being tested. ValueInfo has a type argument which has a TypeProto value. Let me see if comparing the types works better.

Reply:
Updated by comparing classes of TypeProto. Not sure if this is much better, but definitely seems less arbitrary than comparing doc strings.